src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1565,6 +1617,40 @@ void MulticopterPositionControl::control_auto(float dt)
 }
 
 void
+MulticopterPositionControl::do_ground_takeoff(float dt)
+{
+	// check if we are not already in air.
+	// if yes then we don't need a jumped takeoff anymore
+	if (!_takeoff_jumped && !_vehicle_land_detected.landed && !_landed_locked && _takeoff_thrust_sp < FLT_EPSILON) {
+		_takeoff_jumped = true;
+	}
+
+	if (!_takeoff_jumped) {
+		// ramp thrust setpoint up
+		if (_vel(2) > -_params.tko_speed) {
+			_takeoff_thrust_sp += 0.5f * dt;
+			_vel_sp.zero();
+			_vel_prev.zero();
+
+		} else {
+			// copter has reached takeoff speed
+			// we can assume we are off the ground, set takeoff target velocity
+			// set the integator z value to hover thrust since it's the best guess we have
+			_takeoff_jumped = true;
+			_thrust_int(2) = _params.thr_hover;
+			_thrust_int(2) = -math::constrain(_thrust_int(2), _params.thr_min, _params.thr_max);
+			_landed_locked = false;
+		}
+	}
+
+	if (_takeoff_jumped) {
+		// reset previous and current velocity setpoint, now the position controller can take over
+		_vel_sp(2) = -_params.tko_speed;
+		_vel_sp_prev(2) = -_params.tko_speed;
+	}
```

Comment:
still dont get why we set it to tko_speed and not to the current velocity (I am aware the difference might not be significant, but does it not make more sense to the set it to the actual value)